dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java

```diff
@@ -199,6 +201,45 @@ public void testBuildLinkageCheckDependencyGraph_multipleArtifacts() {
         secondElement.getLeaf().getArtifactId());
   }
 
+  @Test
+  public void testBuildVerboseDependencyGraph_singleArtifact() {
+    DependencyGraph graph =
+        dependencyGraphBuilder.buildVerboseDependencyGraph(logging);
+
+    List<DependencyPath> list = graph.list();
+    DependencyPath firstElement = list.get(0);
+    Assert.assertEquals(
+        "commons-logging",
+        firstElement.getLeaf().getArtifactId());
+    Assert.assertEquals(6, list.size()); // optional dependencies are included
```

Comment:
Shouldn't the tests assert that the whole graph is what we expect not just the first node and the amount of nodes?

Reply:
Ideally yes, assuming that's fixed (and I think it is) so you caught me taking a shortcut. That said, I generally write the simplest test I can until something fails (and this did fail at one point) the fix the code until it passes. Then I write a test for something else until the test fails. Here I don't expect adding that test would cause a failure though the only way to be sure of that would be to actually write it. :-)

Reply:
In any case, I added several more asserts here to cover more of the graph. I'm still not testing the exact expected value since that would be quite onerous to setup even for this simple an example. Instead I'm sampling a few know properties of particular parts of the graph.